src/utilcode/clrconfig.cpp

```diff
@@ -297,16 +297,25 @@ BOOL CLRConfig::IsConfigEnabled(const ConfigDWORDInfo & info)
         return FALSE;
 }
 
-
-
 // 
 // Look up a DWORD config value.
 // 
 // Arguments:
-//     * info - see file:../inc/CLRConfig.h for details
-//     
+//     * info - see file:../inc/CLRConfig.h for details.
+//
+//     * useDefaultIfNotSet - if true, fall back to the default value if the value is not set.
+//
+//     * acceptExplicitDefaultFromRegutil - if false, only accept a value returned by REGUTIL if it is
+//           different from the default value. This parameter is useful as a way to preserve existing
+//           behavior.
+//
+//     * result - the result.
+//
+// Return value:
+//     * true for success, false otherwise.
+// 
 // static
-DWORD CLRConfig::GetConfigValue(const ConfigDWORDInfo & info)
+bool CLRConfig::GetConfigValue(const ConfigDWORDInfo & info, bool useDefaultIfNotSet, bool acceptExplicitDefaultFromRegutil, DWORD *result)
```

Comment:
I don't think this method has really been made more complex. In fact, there were some comments in there about how it previously wasn't treating explicitly defined default values coming from REGUTIL properly. Those areas have now been cleaned up, and the switch was added to preserve existing behavior for backwards compatibility.

If we were to change it back, we'd have to always check whether the (JSON) property key was set, and then pass in a ConfigDWORDInfo which will have a different value for the default based on whether the JSON value was set or not. And in the end, we'll be back in the situation where callers to GetConfigValue can't know whether the returned value was actually set or whether it was a default.

I think this change is worth keeping. However, if you feel strongly about it, I'm willing to change it.